bundles/core/src/main/java/com/adobe/cq/commerce/core/components/models/product/Variant.java

```diff
@@ -44,5 +45,9 @@ public interface Variant {
 
     Map<String, Integer> getVariantAttributes();
 
+    default Map<String, String> getVariantAttributesUid() {
```

Comment:
The implementation expects the returned map to be mutable
```
mappedVariant.getVariantAttributesUid().put(option.getCode(), option.getUid().toString());
```
Returning an empty map here will throw an exception with the above statement if not correctly implemented by customers.